### PR TITLE
[6.x] Throw exception on commit without transaction

### PR DESCRIPTION
**Pull request by BobMali**
This PR adresses issue [#29505](https://github.com/laravel/framework/issues/29505).

At the current state a database commit will not commit if no transaction is active. However due to a missing notification this might go unnoticed and lead to some confusion.
With this PR an exception will be thrown in the specified case to notify the developer of a missing transaction.

Due to the change in behaviour of the commit-method this change is considered breaking.

### CODE REVIEW

File: src/Illuminate/Database/Concerns/TransactionException.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Illuminate\Database\Concerns;
+
+use Exception;
+
+class TransactionException extends Exception
+{
+}
```
Comment (crynobone) [inline src/Illuminate/Database/Concerns/TransactionException.php]: A bit odd to have `Exception` under `Concern` namespace, technically this should be under contracts or exceptions folder.

**Comment by mfn** [comment]
This can't go into 6.x, it's breaking change (in behaviour). People accidentally might use this.

See https://github.com/laravel/framework/pull/29693

File: src/Illuminate/Database/Concerns/TransactionException.php (hunk 1 above)
Comment (BobMali) [reply src/Illuminate/Database/Concerns/TransactionException.php]: If i understand it correctly `Contracts` is meant for interfaces only. The `Exception`-namespace was my idea as well, but i did not know where to put it. Seemed still odd in the `Concerns`-namespace.
In the end i used the `Eloquent`-namespace as inspiration and just dropped it where i used it.
Is there any special guideline for exceptions?

**Comment by BobMali** [comment]
So this would be on hold until 7.x?
@taylorotwell what is the correct way to handle this?

File: src/Illuminate/Database/Concerns/TransactionException.php (hunk 1 above)
Comment (crynobone) [reply src/Illuminate/Database/Concerns/TransactionException.php]: https://github.com/illuminate/contracts/search?utf8=%E2%9C%93&q=Exception&type=

There are exceptions in the contracts repository

Comment (BobMali) [reply src/Illuminate/Database/Concerns/TransactionException.php]: Well...yeah. But they are used there, right? So thats the impression i got, that exceptions just are created and dropped where they are used first.
Is there a preferred way in the laravel framework to handle such things?

**Comment by taylorotwell** [comment]
Could be a breaking change for some applications.

**Pull request closed without merging**